### ISSUE TITLE
How to use point cloud data format in 3D convolutional NN?

### ISSUE THREAD

**Issue by xuzhang5788** [Vincentx15/binding-project#1]
Thank you for your repo. 
In your paper, you mentioned that you used point cloud to represent protein binding sites. Could you please tell me what the data format looks like? I mean for each sample, there are some number of points( the number of points will be different for each sample). For each point, its representation should include location (x, y, z) and features ( I think it should be tell what atom it is, such as C, O, etc. using one-hot encoder). Please correct me.

However, in your model, you used, such as "nn.Conv3d(4, 64, kernel_size=3, padding=1)". I think this layer is for voxel representation with 3d grid. I can't understand how to do this for point cloud data format. Because each point can not be perfectly located at the grid.

**Comment by Vincentx15** [comment]
Hello, this is an old project and I must confess I never cleaned it, so it is in pretty bad shape... I will maybe clean it this year if I make a second version.

The data points are voxelized, using a PCA axis alignment as a preprocessing. I planned to include more options, but for now it is just a one-hot encoding of the atom in the grid cell it falls into.

This is done in the script binding_project/protein_binding/shape/data_preprocessor.py, embed_dir_parallel function will turn a directory of pdbs into their corresponding numpy tensors.
